cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/LogAppender.java

```diff
@@ -123,38 +128,62 @@ private void addErrorCodeTags(ILoggingEvent event, Map<String, String> modifiabl
     }
   }
 
-  private void addErrorClassificationTags(ILoggingEvent event, Map<String, String> modifiableMDC) {
+  private void addErrorClassificationTags(ILoggingEvent event, Map<String, String> modifiableMdc) {
     if (event.getThrowableProxy() == null
         || !(event.getThrowableProxy() instanceof ThrowableProxy)) {
       return;
     }
     Throwable throwable = ((ThrowableProxy) (event.getThrowableProxy())).getThrowable();
-    while(throwable != null) {
+    while (throwable != null) {
       if (throwable instanceof WrappedStageException) {
-        modifiableMDC.put(Logging.TAG_FAILED_STAGE,
+        modifiableMdc.put(Logging.TAG_FAILED_STAGE,
             ((WrappedStageException) throwable).getStageName());
       }
+      if (ValidationException.class.getName().equals(throwable.getClass().getName())) {
```

Comment:
So, the problem here is that cdap-etl-api is for communication between the ETL application and plugins. Platform does not and should not know anything about it. If you want platform to get a custom extraciton from the ValidationException the are two ways to do it:
(a) Wrap ValidationException into WrappedStageException
(b)
 * Create an interface in platform API, e.g. FailureStageProvider with `getStageName` method
 * Make both WrappedStageException and ValidationException implement it, moving the code below into ValidationException. 
 * If you need more than just stage name, create an object that has all details, name interface FailureDetailsProvider and make it return such object.
 * Now you don't need dependency on cdap-etl-api here and you can cast to new interface

Reply:
Got it, thanks for the explanation.

Made the changes accordingly.